Makefile

```diff
@@ -8,268 +28,404 @@ usage:
 	# also quite time consuming. So we want to verify that the rules still
 	# work, but don't build them for every board.
 	#
 	# The choice of building for the nrf52dk was chosen by random die roll.
-	@printf "$$(tput bold)*****************************$$(tput sgr0)\n"
-	@printf "$$(tput bold)* CI: Debug Support Targets *$$(tput sgr0)\n"
-	@printf "$$(tput bold)*****************************$$(tput sgr0)\n"
 	@CI=true $(MAKE) -C boards/nordic/nrf52dk lst
 	@CI=true $(MAKE) -C boards/nordic/nrf52dk debug
 	@CI=true $(MAKE) -C boards/nordic/nrf52dk debug-lst
 
-.PHONY: ci-documentation
-ci-documentation:
-	@printf "$$(tput bold)*********************$$(tput sgr0)\n"
-	@printf "$$(tput bold)* CI: Documentation *$$(tput sgr0)\n"
-	@printf "$$(tput bold)*********************$$(tput sgr0)\n"
-	@CI=true tools/toc.sh
-
+.PHONY: ci-job-collect-artifacts
+ci-job-collect-artifacts: ci-job-compilation
+	# Collect binary images for each board
+	#
+	# This is currently used only for code size detection changes, but in
+	# the future may also be used to support checks for deterministic builds.
+	@mkdir -p ./ci-artifacts
+	@rm -rf "./ci-artifacts/*"
+	@for f in $$(find ./target -iname '*.bin' | grep -E "release/.*\.bin");\
+		do mkdir -p "ci-artifacts/$$(dirname $$f)";\
+		cp "$$f" "ci-artifacts/$$f";\
+		done
 
-## Actual Rules (Netlify)
 
-.PHONY: ci-rustdoc
-ci-rustdoc:
-	@printf "$$(tput bold)*****************************$$(tput sgr0)\n"
-	@printf "$$(tput bold)* CI: Rustdoc Documentation *$$(tput sgr0)\n"
-	@printf "$$(tput bold)*****************************$$(tput sgr0)\n"
-	@#n.b. netlify calls tools/netlify-build.sh, which is a wrapper
-	@#     that first installs toolchains, then calls this.
-	@tools/build-all-docs.sh
-
-## End CI rules
-##
-###################################################################
 
+### ci-runner-github-tests jobs:
+.PHONY: ci-job-libraries
+ci-job-libraries:
+	$(call banner,CI-Job: Libraries)
+	@cd libraries/enum_primitive && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/riscv-csr && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/tock-cells && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/tock-register-interface && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/tock-rt0 && CI=true RUSTFLAGS="-D warnings" cargo test
 
-.PHONY: audit
-audit:
-	@for f in `./tools/list_lock.sh`; do echo "$$(tput bold)Auditing $$f"; (cd "$$f" && cargo audit || exit 1); done
+.PHONY: ci-job-archs
+ci-job-archs:
+	$(call banner,CI-Job: Archs)
+	@for arch in `./tools/list_archs.sh`;\
+		do echo "$$(tput bold)Test $$arch";\
+		cd arch/$$arch;\
+		CI=true RUSTFLAGS="-D warnings" TOCK_KERNEL_VERSION=ci_test cargo test || exit 1;\
+		cd ../..;\
+		done
+
+.PHONY: ci-job-kernel
+ci-job-kernel:
+	$(call banner,CI-Job: Kernel)
+	@cd kernel && CI=true RUSTFLAGS="-D warnings" TOCK_KERNEL_VERSION=ci_test cargo test
 
-.PHONY: emulation-setup
-emulation-setup: SHELL:=/usr/bin/env bash
```

Comment:
I think this line is required for the test to run correctly

Reply:
I just removed the `mktemp`, same effect